docs/tutorials/customizing-environments.md

```diff
@@ -1,151 +1,180 @@
+title: Customizing the shell environment
+description: Using setup scripts, aliases, and environment variables to improve your workflows.
 ---
 
-# Customizing the environment shell hook
+# Customizing the shell environment
 
-This guide uses the [environment's][environment_concept]
-[shell hooks][hook_concept] to set up a PostgreSQL development database.
+Activating a Flox [environment][environment_concept] places you into a subshell.
+You likely already have some customizations built into your shell from your shell's configuration (`.bashrc`, `.zshrc`, `config.fish`, etc), but it can be convenient to further customize your shell based on the project that you're working on.
+This guide will walk you through leveraging various features of a Flox environment to improve your quality of life when developing a Rust project, but many of the ideas are applicable to other languages as well.
 
-## Create a PostgreSQL environment
+## Setup
 
-Say your project has a variable it expects to be set and you need to generate
-the value for development.
-This is a great use for the environment's **[shell hooks][hook_concept]**.
+Let's assume you have a Rust project that you regularly work on.
+To do that work you would already have `cargo`, `rustc`, and a few other tools installed.
+For more details on what it looks like to develop in Rust with Flox, see the [Rust language guide][rust_guide].
 
-Let's set up a Flox [environment][environment_concept] using the `postgresql_15`
-package in your environment:
+If you'd like to follow along with a real Flox environment, create an environment via [`flox init`][flox_init] and install the tools as shown below:
 
-``` console
-$ flox init --name postgres-example
+```bash
+$ mkdir mycli
+$ cd mycli
+$ flox init
+$ flox install rustc cargo libiconv
+```
 
-✨ Created environment postgres-example (aarch64-darwin)
+Then generate a basic "Hello, World" program using `cargo init` inside the environment:
 
-Next:
-  $ flox search <package>    <- Search for a package
-  $ flox install <package>   <- Install a package into an environment
-  $ flox activate            <- Enter the environment
+```bash
+$ flox activate -- cargo init --bin .
 ```
 
-``` console
-$ flox install postgresql_15
-✅ 'postgresql_15' installed to environment postgres-example at /Users/youruser
-```
+## Vars, hook, or profile?
 
-## Customize the environment's shell hook
+When customizing your shell environment you have three basic knobs you can turn:
 
-Let's add some properties PostgreSQL needs to run properly in this
-[environment][environment_concept].
+- The `[vars]` section
+- The `hook.on-activate` script
+- The `[profile]` section
 
-``` console
-$ flox edit
-```
+The logic for deciding where a customization should go is application specific, but there are some simple guidelines you can follow.
+For a full discussion of what logic to place in which section and why, see the [activation concept page][activation_concept].
+Otherwise, try this:
```

Comment:
I like the idea of logic to explain this but I don't think the bulleted list format is working here. Perhaps it's just that the indentation didn't work as you intended?

Reply:
Oh that definitely got mangled